### PR TITLE
Added support for undefined behaviour sanitizer on native_posix

### PR DESCRIPTION
**Pull request by vanwinkeljan**
Added toolchain abstraction to support undefined behaviour sanitizer on native_posix.

### CODE REVIEW

**Review by SebastianBoe — APPROVED** [review]

File: subsys/debug/Kconfig [hunk 1]
```diff
@@ -54,6 +55,15 @@ config ASAN
 	  compiler with the ``-fsanitize=address`` command line option, and
 	  the libasan library.
 
+config UBSAN
+	bool "Build with undefined behavior sanitizer"
+	depends on ARCH_POSIX
+	help
+	  Builds Zephyr with Undefined Behavior Sanitizer enabled.
+	  This is currently only supported by the native_posix port, and requires
```
Comment (aescolar) [inline subsys/debug/Kconfig]: Just a minor nit:
It is supported by all posix arch boards, not just native_posix. could you rephrase
"This is currently only supported by the native_posix port, and requires"
to
"This is only supported by boards based on the posix arch, and requires"

**Review by aescolar — COMMENTED** [review]
I would have already merged it if it wasn't for that minor thing in the description.

File: subsys/debug/Kconfig (hunk 1 above)
Comment (vanwinkeljan) [reply subsys/debug/Kconfig]: Updated description and took the freedom to lineup the ASAN description with the one used for UBSAN

**Review by aescolar — APPROVED** [review]
